import "galaxy.proto";

package baidu.galaxy;

option cc_generic_services = true;

enum ProcessStatus {
    kProcessRunning = 0;
    kProcessTerminate = 1;
}

message ProcessInfoCheckpoint {
    repeated ProcessInfo process_infos = 1;
}

message ProcessInfo {
    optional string key = 1;
    optional int32 pid = 2; 
    optional ProcessStatus status = 3;
    optional int32 exit_code = 4;
}

message GetProcessStatusRequest {
    optional string key = 1;
}

message GetProcessStatusResponse {
    optional ProcessInfo process = 1;
    optional Status status = 2;
}

message ExecuteRequest {
    optional string key = 1; 
    optional string commands = 2;
    repeated string envs = 3;
    optional string path = 4;
    optional string cgroup_path = 5;
}

message ExecuteResponse {
    optional string key  = 1;
    optional int32 pid = 2;
    optional Status status = 3;
}

message MountPathRequest {
    optional string volume_src = 1;
    optional string volume_dest = 2;
}

message MountPathResponse {
    optional Status status = 2;
}

message UMountPathRequest {
    optional string volume_dest = 1;
}

message UMountPathResponse {
    optional Status status = 2;
}

message InitdHeartBeatRequest {
}

message InitdHeartBeatResponse {
    optional Status status = 1;
}

service Initd {
    // deprecated
    rpc GetProcessStatus(GetProcessStatusRequest) returns (GetProcessStatusResponse);
    // deprecated
    rpc Execute(ExecuteRequest) returns (ExecuteResponse);
    rpc MountPath(MountPathRequest) returns (MountPathResponse);    
    rpc UMountPath(UMountPathRequest) returns (UMountPathResponse);

    rpc InitdHeartBeat(InitdHeartBeatRequest) returns (InitdHeartBeatResponse);
}
